test/view/cartesian_product/at.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright Louis Dionne 2013-2016
+// Copyright Jason Rice 2017
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+
+#include <boost/hana/assert.hpp>
+#include <boost/hana/at.hpp>
+#include <boost/hana/equal.hpp>
+#include <boost/hana/integral_constant.hpp>
+#include <boost/hana/view.hpp>
+
+#include <laws/base.hpp>
+#include <support/seq.hpp>
+namespace hana = boost::hana;
+using hana::test::ct_eq;
+
+
+int main() {
+    auto container = ::seq;
+
```

Comment:
This is missing a unit test for an empty sequence, and for a non-empty sequence containing empty sequences (first, second, third, etc..). Please add a few test cases like that. It's not so much because I doubt the correctness of your implementation, but these sort of test cases are very good at making strange bugs surface (e.g. the unary copy-constructors for tuple surfaces several SFINAE-related bugs).